### ISSUE TITLE
Define loss as classes instead of functions

### ISSUE THREAD

**Issue by mathpluscode** [DeepRegNet/DeepReg#567]
## Subject of the issue

Now loss are functions, we should define them as py classes inheritated from tf.keras.loss

https://github.com/tensorflow/tensorflow/blob/v2.3.1/tensorflow/python/keras/losses.py#L47

Then, we use registry to register our loss classes
